src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagEntryValidator.cs

```diff
@@ -0,0 +1,170 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Linq;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.Query;
+
+namespace Microsoft.Health.Dicom.Core.Features.CustomTag
+{
+    public class CustomTagEntryValidator : ICustomTagEntryValidator
```

Comment:
I am wondering if this is the right pattern to validate the input. Did you look at https://docs.microsoft.com/en-us/aspnet/core/mvc/models/validation?view=aspnetcore-5.0? There is validationAttributes and IValidationObjects you can use to validate each input attribute

Reply:
I see we also validate the input on Core level(https://github.com/microsoft/dicom-server/blob/dcc067d7b2628e68cca9bebfbd98fe8aa5505746/src/Microsoft.Health.Dicom.Core/Features/Retrieve/RetrieveTransferSyntaxHandler.cs#L57). Wondering  what's the best practice here?

Reply:
I tried the model validation, I think it also works. just not very sure should we implement the validation in API layer or Core layer.  I have several consideration:
If we implement the model validation, since it  bind to Asp.net Core MVC, the API layer likely to be the best place to put it
2. In the past, we moved the validation on accepted transfersyntax from API layer to Core (Update GetFrame to support transfer syntax as discussed (#228) · microsoft/dicom-server@ce73611 (github.com), should we follow same thing this time?